### DISCUSSION TITLE
DBeaver behind proxy - update not working

### DISCUSSION THREAD

**Discussion by NullpointerForever** [dbeaver/dbeaver#627] [Q&A] [answered]
Hi,

I have DBeaver Standalone EE behind proxy and update is not working:

```
Unable to connect to repository http://dbeaver.jkiss.org/update/latest/content.xml
Connection timed out: connect
```

I've try set my proxy settings by adding lines to dbeaver.ini:

```
-vmargs
-Dhttp.proxyPort=...
-Dhttp.proxyHost=...
-Dhttp.nonProxyHosts=localhost|127.0.0.1
```

or:

```
-vmargs
-Dorg.eclipse.ecf.provider.filetransfer.excludeContributors=org.eclipse.ecf.provider.filetransfer.httpclient
-Dhttp.proxyPort=...
-Dhttp.proxyHost=...
-Dhttp.nonProxyHosts=localhost|127.0.0.1
```

and there are still problems. Proxy settings in "Database / Drivers" for downloading drivers is working.

Regards

**Comment by serge-rider** [comment, marked as answer]
P2 update isn't yet fully functional in DBeaver.
Eclipse has its own proxy configuration which is also used by P2 update.
But we need to add a few extra plugins to bring this config in preferences.
